src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -281,85 +415,73 @@ public void smokeTest() throws Exception {
-    }
-
-    private void runOnDocker(DumbSlave s) throws Exception {
-        runOnDocker(s, 10);
+    /**
+     * Checks if the golang binary outputs to stdout under normal shell execution.
+     * The binary must NOT output to stdout or else it will crash when Jenkins is terminated
+     * unexpectedly.
+     */
+    @Test public void stdout() throws Exception {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        TeeOutputStream teeOut = new TeeOutputStream(baos, System.out);
+        StreamTaskListener stdoutListener = new StreamTaskListener(teeOut, Charset.defaultCharset());
+        String script = String.format("echo hello world");
+        Controller c = new BourneShellScript(script).launch(new EnvVars(), ws, launcher, stdoutListener);
+        awaitCompletion(c);
+        assertThat(baos.toString(), isEmptyString());
+        c.cleanup(ws);
     }
 
-    private void runOnDocker(DumbSlave s, int sleepSeconds) throws Exception {
-        j.jenkins.addNode(s);
-        j.waitOnline(s);
-        FilePath dockerWS = s.getWorkspaceRoot();
-        Launcher dockerLauncher = s.createLauncher(listener);
-        String script = String.format("echo hello world; sleep %s", sleepSeconds);
-        Controller c = new BourneShellScript(script).launch(new EnvVars(), dockerWS, dockerLauncher, listener);
-        while (c.exitStatus(dockerWS, dockerLauncher, listener) == null) {
-            Thread.sleep(100);
-        }
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        c.writeLog(dockerWS, baos);
-        assertEquals(0, c.exitStatus(dockerWS, dockerLauncher, listener).intValue());
-        assertTrue(baos.toString().contains("hello world"));
-        c.cleanup(dockerWS);
+    private boolean noZombies() throws InterruptedException {
+        ByteArrayOutputStream baos = null;
         do {
             Thread.sleep(1000);
             baos = new ByteArrayOutputStream();
             try {
-                assertEquals(0, dockerLauncher.launch().cmds("ps", "-e", "-o", "pid,stat,comm").stdout(new TeeOutputStream(baos, System.out)).join());
+                assertEquals(0, launcher.launch().cmds("ps", "-e", "-o", "pid,ppid,stat,comm").stdout(new TeeOutputStream(baos, System.out)).join());
             } catch (IOException x) { // no ps? forget this check
                 System.err.println(x);
                 break;
             }
         } while (baos.toString().contains(" sleep "));
-        assertThat("no zombies running", baos.toString(), not(containsString(" Z ")));
-        s.toComputer().disconnect(new OfflineCause.UserCause(null, null));
+        return !baos.toString().contains(" Z ");
+    }
+
+    // to assist with debugging
+    private String psOut() throws InterruptedException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+            launcher.launch().cmds("ps", "-ef").stdout(new TeeOutputStream(baos, System.out)).join();
```

Comment:
How noisy is this? Maybe better to delete all of the calls in the tests but leave the function here in case someone wants to use it while investigating a test failure? Would be good to update the comment above the function to explain what it does in addition to noting that it is useful for debugging.

Reply:
Will do.  It turns out it's not very noisy for the docker tests, but for native, it is _quite_ noisy.